yt/frontends/boxlib/io.py

```diff
@@ -145,15 +145,24 @@ def _read_particle_fields(self, chunks, ptf, selector):
                         rdata = np.fromfile(
                             f, pheader.real_type, pheader.num_real * npart
                         )
-                        x = np.asarray(rdata[0 :: pheader.num_real], dtype=np.float64)
-                        y = np.asarray(rdata[1 :: pheader.num_real], dtype=np.float64)
-                        if g.ds.dimensionality == 2:
-                            z = np.ones_like(y)
-                            z *= 0.5 * (g.LeftEdge[2] + g.RightEdge[2])
-                        else:
-                            z = np.asarray(
-                                rdata[2 :: pheader.num_real], dtype=np.float64
-                            )
+
+                        # Allow reading particles in 1, 2, and 3 dimensions,
+                        # setting the appropriate default for unused dimensions.
+                        pos = []
```

Comment:
Since `pos[0]` is used as a default for missing dimensions, I suggest adding an assertion to check that dimensionality is sane, in case someday it's not and the bug isn't as obvious as possible.
```python
assert self.dimensionality >= 1
```

Another solution would be to use default values that do not depend on the results of the first iteration (my prefered approach)

Reply:
Thanks for your comment. Does my change address your concern?

Reply:
It does, thank you !
Now I just noticed that you're looping on 1, 2, 3 but since you're really using 0-based indices I think the code could be simplified.

Reply:
Yes, that was intentional, looping over 1, 2, and 3 dimensions. It seemed more clear me what the loop was doing at the expense of slightly less simplified code.

Reply:
Not critical, clarity-in-the-eyes-of-the-author wins here 😄

Reply:
Thanks for merging!